tests/scalers/aws/aws_cloudwatch_min_metric_value/aws_cloudwatch_min_metric_value_test.go

```diff
@@ -0,0 +1,188 @@
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  maxReplicaCount: 2
+  minReplicaCount: 0
+  cooldownPeriod: 1
+  triggers:
+    - type: aws-cloudwatch
+      authenticationRef:
+        name: keda-trigger-auth-aws-credentials
+      metadata:
+        awsRegion: {{.AwsRegion}}
+        namespace: {{.CloudWatchMetricNamespace}}
+        dimensionName: {{.CloudWatchMetricDimensionName}}
+        dimensionValue: {{.CloudWatchMetricDimensionValue}}
+        metricName: {{.CloudWatchMetricName}}
+        targetMetricValue: "1"
+        minMetricValue: "1"
+        metricCollectionTime: "120"
+        metricStatPeriod: "60"
+`
+)
+
+var (
+	testNamespace                  = fmt.Sprintf("%s-ns", testName)
+	deploymentName                 = fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName               = fmt.Sprintf("%s-so", testName)
+	secretName                     = fmt.Sprintf("%s-secret", testName)
+	cloudwatchMetricName           = fmt.Sprintf("cw-%d", GetRandomNumber())
+	awsAccessKeyID                 = os.Getenv("TF_AWS_ACCESS_KEY")
+	awsSecretAccessKey             = os.Getenv("TF_AWS_SECRET_KEY")
+	awsRegion                      = os.Getenv("TF_AWS_REGION")
+	cloudwatchMetricNamespace      = "DoesNotExist"
+	cloudwatchMetricDimensionName  = "dimensionName"
+	cloudwatchMetricDimensionValue = "dimensionValue"
+	minReplicaCount                = 0
+	minMetricValueReplicaCount     = 1
+)
+
+// This test is to verify that the scaler returns the minMetricValue when the metric
+// value is null and ignoreNullValues is set to true.
+func TestCloudWatchScalerWithMinMetricValue(t *testing.T) {
+	ctx := context.Background()
+
+	// setup cloudwatch
+	cloudwatchClient, err := cloudwatch.NewClient(ctx, awsRegion, awsAccessKeyID, awsSecretAccessKey, "")
+	assert.Nil(t, err, "error creating cloudwatch client")
+
+	// check that the metric in question is not already present, and is returning
+	// an empty set of values.
+	metricQuery := cloudwatch.CreateMetricDataInputForEmptyMetricValues(cloudwatchMetricNamespace, cloudwatchMetricName, cloudwatchMetricDimensionName, cloudwatchMetricDimensionValue)
+	metricData, err := cloudwatch.GetMetricData(ctx, cloudwatchClient, metricQuery)
+	require.Nil(t, err, "error getting metric data")
+	require.Nil(t, cloudwatch.ExpectEmptyMetricDataResults(metricData), "metric data should be empty")
+
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData()
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+	defer DeleteKubernetesResources(t, testNamespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 1),
+		"replica count should be %d after 1 minute", minReplicaCount)
+
+	// check that the deployment scaled up to the minMetricValueReplicaCount
+	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, minMetricValueReplicaCount, 60)
```

Comment:
I'm not 100% if this will work or it should be 
```suggestion
	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 1),
		"replica count should be %d after 1 minute", minMetricValueReplicaCount)

	// check that the deployment scaled up to the minMetricValueReplicaCount
	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, minMetricValueReplicaCount, 60)
```

Basically current test is checking the the workload is 0 and then it doesn't change from 1.